### PR TITLE
release.nix: add platform filtering

### PR DESCRIPTION
**Pull request by yorickvP**

### CODE REVIEW

File: release.nix [hunk 1]
```diff
@@ -22,29 +22,39 @@ let
...
+disciplina-pkgs = rec {
+  inherit disciplina-wallet;
+
+  disciplina-wallet-flatpak = (buildFlatpak {
     app-id = "io.disciplina.Wallet";
     command = disciplina-wallet-wrapped;
     finish-args = [ "--share=network" ];
-  };
+  }).overrideAttrs (old: {
+    meta.platforms = lib.platforms.linux;
```
Comment (yegortimoshenko) [inline release.nix]: Let's have `nix-flatpak` generate that.

File: release.nix [hunk 2]
```diff
@@ -22,29 +22,39 @@ let
...
-  '';
-
-  disciplina-wallet-macos-app = buildMacOSApp {
+  disciplina-wallet-macos-app = (buildMacOSApp {
     name = "Disciplina";
     target = disciplina-wallet-macos-sandbox;
     withOpen = true;
-  };
+  }).overrideAttrs (old: {
+    meta.platforms = lib.platforms.darwin;
```
Comment (yegortimoshenko) [inline release.nix]: Let's have `nix-macos-app` generate that :-)

File: release.nix [hunk 3]
```diff
@@ -22,29 +22,39 @@ let
...
+  });
+};
+in
+disciplina-pkgs // {
+  all-buildable = with lib; let
+      system = lib.systems.elaborate { inherit (pkgs) system; };
+      match = lib.meta.platformMatch system;
+    in filterAttrs (name: pkg: any match pkg.meta.platforms) disciplina-pkgs;
```
Comment (yegortimoshenko) [inline release.nix]: I'm not sure there's a point in making `release.nix` fully buildable, that's why I moved this stuff to `release.nix` in the first place.

Comment (yorickvP) [reply release.nix]: I want to be able to do `nix build -f ./release.nix all-buildable`, that's why

**Pull request merged**